src/pubkey.cpp

```diff
@@ -4,19 +4,182 @@
 
 #include "pubkey.h"
 
-#include "eccryptoverify.h"
+#include <secp256k1.h>
+#include <secp256k1_recovery.h>
 
-#include "ecwrapper.h"
+namespace
+{
+secp256k1_context* secp256k1_context_verify = NULL;
```

Comment:
Only the secp256k1_context_randomize (not used in this module) and secp256k1_context_destroy (obviously) functions need exclusive access, so no, as long as all uses happen after the context is guaranteed to be constructed.